python/ray/data/datasource/datasource.py

```diff
@@ -18,20 +18,21 @@
 
 @PublicAPI
 class Datasource:
-    """Interface for defining a custom ``ray.data.Dataset`` datasource.
+    """Interface for defining a custom :class:`~ray.data.Dataset` datasource.
 
-    To read a datasource into a dataset, use ``ray.data.read_datasource()``.
-    To write to a writable datasource, use ``Dataset.write_datasource()``.
+    To read a datasource into a dataset, use :meth:`~ray.data.read_datasource`.
+    To write to a writable datasource, use :meth:`~ray.data.Dataset.write_datasource`.
 
-    See ``RangeDatasource`` and ``DummyOutputDatasource`` for examples
+    See :class:`~ray.data.datasource.RangeDatasource` and
+    :class:`~ray.data.datasource.DummyOutputDatasource` for examples
     of how to implement readable and writable datasources.
 
-    Datasource instances must be serializable, since ``create_reader()`` and
-    ``write()`` are called in remote tasks.
+    Datasource instances must be serializable, since :meth:`~ray.data.Datasource.create_reader` and
+    :meth:`~ray.data.Datasource.write` are called in remote tasks.
```

Comment:
Maybe put this in a `.. note::` since this is more advanced and move below "For an example of subclassing..."?